velox/connectors/hive/HiveConfig.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/core/Context.h"
+
+#include <boost/algorithm/string.hpp>
+
+namespace facebook::velox::connector::hive {
+
+/// Hive connector configs
+class HiveConfig {
+ public:
+  enum class InsertExistingPartitionsBehavior {
+    kAppend,
+    kError,
+    kOverwrite,
+  };
+
+  /// Can new data be inserted into existing partitions or existing
+  /// unpartitioned tables.
+  static constexpr const char* FOLLY_NONNULL kImmutablePartitions =
+      "hive.immutable-partitions";
+
+  /// Maximum number of partitions per writer.
+  static constexpr const char* FOLLY_NONNULL kMaxPartitionsPerWriters =
+      "hive.max-partitions-per-writers";
+
+  /// Default value for insert existing partitions behavior.
+  static constexpr const char* FOLLY_NONNULL kInsertExistingPartitionsBehavior =
+      "hive.insert-existing-partitions-behavior";
+
+  static bool isImmutablePartitions(const Config* FOLLY_NONNULL baseConfig) {
+    return baseConfig->get<bool>(kImmutablePartitions, true);
+  }
+
+  static uint32_t maxPartitionsPerWriters(
+      const Config* FOLLY_NONNULL baseConfig) {
+    return baseConfig->get<uint32_t>(kMaxPartitionsPerWriters, 100);
+  }
+
+  static InsertExistingPartitionsBehavior insertExistingPartitionsBehavior(
```

Comment:
To validate and set the value, replicating logic https://github.com/prestodb/presto/blob/ff412ef4b745eb31549cc0abe25e885a63875f58/presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java#L643-L650.